### PR TITLE
implement ClearRow() query

### LINKED ISSUE #1240: implement delete row
### Success criteria (What criteria will consider this ticket closeable?)
You will have the ability to remove an existing row from pilosa

### PR DESCRIPTION
**Pull request by travisturner**
## Overview

This PR implements `ClearRow()`.

Fixes #1240 

TODO:
- [x] Tests
- [x] Documentation
- [x] Prevent `ClearRow()` on `int` fields.
- [x] For `date` fields, execute `ClearRow()` on every view.
- [x] For `mutex` fields, must ensure the `vector` isn't storing information that would need to be cleared.

## Pull request checklist

- [x] I have read the [contributing guide](https://github.com/pilosa/pilosa/blob/master/CONTRIBUTING.md).
- [x] I have agreed to the [Contributor License Agreement](https://cla-assistant.io/pilosa/pilosa).
- [x] I have updated the [documentation](https://github.com/pilosa/pilosa/tree/master/docs).
- [x] I have resolved any merge conflicts.
- [x] I have included tests that cover my changes.
- [x] All new and existing tests pass.

## Code review checklist
This is the checklist that the reviewer will follow while reviewing your pull request. You do not need to do anything with this checklist, but be aware of what the reviewer will be looking for.

- [ ] Ensure that any changes to external docs have been included in this pull request.
- [ ] If the changes require that minor/major versions need to be updated, tag the PR appropriately.
- [ ] Ensure the new code is [properly commented](https://github.com/golang/go/wiki/CodeReviewComments#doc-comments) and follows [Idiomatic Go](https://dmitri.shuralyov.com/idiomatic-go).
- [ ] Check that tests have been written and that they cover the new functionality.
- [ ] Run tests and ensure they pass.
- [ ] Build and run the code, performing any applicable integration testing.

### CODE REVIEW

**Comment by travisturner** [comment]
@jaffee you're right. That's why I have those listed in the TODOs of this WIP PR.

**Comment by jaffee** [comment]
ha! I think my brain sees checkboxes and assumes it's the standard PR checklist. will read more closely in the future :)

**Comment by travisturner** [comment]
@jaffee @tgruben I removed the WIP from this PR.

**Comment by travisturner** [comment]
I added a commit which modifies the ranked cache implementation to allow `cache.Add(row, 0)` in order to remove a row from the cache.

File: fragment.go [hunk 1]
```diff
@@ -2182,6 +2221,12 @@ func pos(rowID, columnID uint64) uint64 {
 type vector interface {
 	Get(colID uint64) (uint64, bool)
 	Set(colID, rowID uint64)
+	// TODO: Clear(colID, rowID uint64)
```
[the remark this answers is not in the document]
Comment (travisturner) [reply fragment.go]: The only time you need `Set()` or `Clear()` is if the implementation of the `vector` interface stores values separate from fragment storage (for example, if one were to use a map). Currently, the only `vector` implementations in Pilosa use fragment storage directly, so there's no need to call `Set()` or `Clear()` on those. In fact, we should probably just remove the `Set()` method at this point.

Also, I think we should add an `error` to the `Get()` method to allow that lookup to return any errors associated with invalid data (for example, when the `Get()` finds more than one row associated to a column in a `mutex` field.

File: executor.go [hunk 2]
```diff
@@ -1137,6 +1139,76 @@ func (e *executor) executeClearBitField(ctx context.Context, index string, c *pq
...
+	// Ensure the field type supports ClearRow().
+	fieldName, err := c.FieldArg()
+	if err != nil {
+		return false, errors.New("ClearRow() argument required: field")
+	}
+	field := e.Holder.Field(index, fieldName)
+	if field == nil {
+		return false, ErrFieldNotFound
+	}
+	if field.Type() == FieldTypeInt {
```
[the remark this answers is not in the document]
Comment (travisturner) [reply executor.go]: Makes sense. I'll change it.

**Pull request merged**